plugins/inventory/hrobot.py

```diff
@@ -0,0 +1,180 @@
+        super(InventoryModule, self).parse(inventory, loader, path)
+        servers = {}
+        config = self._read_config_data(path)
+        self.load_cache_plugin()
+        cache_key = self.get_cache_key(path)
+
+        # cache may be True or False at this point to indicate if the inventory is being refreshed
+        # get the user's cache option too to see if we should save the cache if it is changing
+        user_cache_setting = self.get_option('cache')
+
+        # read if the user has caching enabled and the cache isn't being refreshed
+        attempt_to_read_cache = user_cache_setting and cache
+        # update if the user has caching enabled and the cache is being refreshed; update this value to True if the cache has expired below
+        cache_needs_update = user_cache_setting and not cache
+
+        # attempt to read the cache if inventory isn't being refreshed and the user has caching enabled
+        if attempt_to_read_cache:
+            try:
+                servers = self._cache[cache_key]
+            except KeyError:
+                # This occurs if the cache_key is not in the cache or if the cache_key expired, so the cache needs to be updated
+                cache_needs_update = True
+        else:
+            servers = self.get_servers()
+
+        if cache_needs_update:
+            servers = self.get_servers()
+
+            # set the cache
+            self._cache[cache_key] = servers
+
+        self.populate(servers)
+
+    def populate(self, servers):
+        filters = self.get_option("filters")
+        for server in servers:
+            s = server["server"]
+
+            matched = self.filter(s, filters)
+            if matched:
+                self.inventory.add_host(s["server_name"])
+                self.inventory.set_variable(s["server_name"], 'ansible_host', s['server_ip'])
+                self.inventory.set_variable(s["server_name"], 'server_ip', s['server_ip'])
+                self.inventory.set_variable(s["server_name"], 'product', s['product'])
+
+                self.inventory.set_variable(s["server_name"], 'dc', s['dc'])
+                self.inventory.set_variable(s["server_name"], 'traffic', s['traffic'])
+                self.inventory.set_variable(s["server_name"], 'flatrate', s['flatrate'])
+                self.inventory.set_variable(s["server_name"], 'status', s['status'])
+
+                strict = self.get_option('strict')
+                self._add_host_to_keyed_groups(
+                    self.get_option("keyed_groups"), {}, s["server_name"], strict=strict
+                )
+
+    def filter(self, server, filters):
+        matched = True
+        for key, value in filters.items():
+            if server[key] != value:
+                matched = False
+                break
+        return matched
+
+    def get_servers(self):
+        user = self.get_option("hrobot_api_user")
+        password = self.get_option("hrobot_api_password")
```

Comment:
To mirror the above changes:
```suggestion
        user = self.get_option("hetzner_user")
        password = self.get_option("hetzner_password")
```
I think we should add a function similar to `fetch_url_json` in `module_utils` which works for plugins instead of modules, with the code from this function as a basis.